bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -56,6 +56,24 @@ public PageChangeListener(String sitemapName, String pageId, ItemUIRegistry item
         this.sitemapName = sitemapName;
         this.pageId = pageId;
         this.itemUIRegistry = itemUIRegistry;
+
+        updateItemsAndWidgets(widgets);
+    }
+
+    private void updateItemsAndWidgets(EList<Widget> widgets) {
+        if (this.widgets != null) {
+            // cleanup statechange listeners in case widgets were removed
+            items = getAllItems(this.widgets);
+            for (Item item : items) {
+                if (item instanceof GenericItem) {
+                    ((GenericItem) item).removeStateChangeListener(this);
+                    ;
+                } else if (item instanceof GroupItem) {
```

Comment:
hmmm. Just realizing: `GroupItem` extends `GenericItem`. So I don't see a need for this second part. Do you? 
(it's wrong below too, which is probably where you copied it from...?)